src/components/popover/popover.js

```diff
@@ -0,0 +1,210 @@
+(function() {
+'use strict';
+
+/**
+ * @ngdoc module
+ * @name material.components.popover
+ */
+angular.module('material.components.popover', [
+  'material.core'
+])
+  .directive('mdPopover', MdPopoverDirective);
+
+/**
+ * @ngdoc directive
+ * @name mdPopover
+ * @module material.components.popover
+ * @description
+ * Popovers are used to describe elements that are interactive and either graphical or textual.
+ *
+ * Place a `<md-popover>` as a child of the element it relates to.
+ *
+ * A popover will activate when the user clicks the parent.
+ *
+ * @usage
+ * <hljs lang="html">
+ * <md-icon icon="/img/icons/ic_play_arrow_24px.svg">
+ *   <md-popover>
+ *     Play Music
+ *   </md-popover>
+ * </md-icon>
+ * </hljs>
+ *
+ * @param {expression=} md-visible Boolean bound to whether the popover is currently visible.
+ * @param {expression=} md-placement string bound to location of popover.
+ */
+function MdPopoverDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdTheming, $rootElement) {
+
+  var POPOVER_SHOW_DELAY = 400;
+  var POPOVER_WINDOW_EDGE_SPACE = 8;
+
+  return {
+    restrict: 'E',
+    transclude: true,
+    template:
+      '<div class="md-background"></div>' +
+      '<div class="md-content" ng-transclude></div>',
+    scope: {
+      visible: '=?mdVisible',
+      placement: '=mdPlacement'
```

Comment:
Functionally it's optional and working but I'll add the ? to make it more obvious for people